### DISCUSSION TITLE
Issue Pertaining to Mesh File Used in WRF-CTSM Coupled Model

### DISCUSSION THREAD

**Discussion by CarterEWatson** [esmf-org/esmf#266] [Remapping / Regridding] [answered]
### Requirements

- [X] Reviewed [ESMF Reference Manual](https://earthsystemmodeling.org/doc/)
- [X] Searched [GitHub Discussions](https://github.com/orgs/esmf-org/discussions?discussions_q=)

### Affiliation(s)

University of Georgia

### ESMF Version

v8.6.1

### Issue

Good afternoon everyone,

I am having a rather frustrating issue with an ESMF mesh file (it won't let me attach NetCDF files, but I can find some way to send it if needed) that I believe you all might be able to help me solve. For starters, I am attempting to use the Weather Research and Forecasting (WRF) model coupled with the Community Terrestrial Systems Model (CTSM). Required by this are multiple ESMF mesh files, the one in question has already been utilized to create input data for the model (therefore, I know for certain that the file itself is not corrupt.) Whenever I begin the process to start running the model, I get the following error related to the mesh file:

_(lilac_atmcap_init):Mesh for atmosphere is created for /opt/home/cwat/WRF/CTSM/tools/site_and_regional/esmfmeshfile_d01.nc
ERROR: lilac_atmcap: n, lon, mesh_lon = 1 268.4401855469 -91.5598144531
ERROR: Unknown error submitted to shr_abort_abort._

To the best of my understanding, the error suggests that the mesh file has longitudes recorded from -180° to 180°, while my model expects them to be from 0° to 360°. It would be amazing if you all are able to provide some guidance (i.e., is this an ESMF issue? Is there an ESMF regrinding tool or a simple python or ncl script that can modify the mesh file appropriately? Have you all seen an error similar to this previously?) I have reached out to the UCAR support staff for CTSM, but they are not able to help with this problem. Additionally, LILAC does not have a dedicated support team, so I do not know with whom I should be speaking with. So, if this is not an appropriate place to solicit this answer, I do apologize.

### Autotag

@oehmke

**Comment by oehmke** [comment, marked as answer]
Hi Carter, Sorry for the delay in answering. A lot of us have been on vacation. This isn't an ESMF problem in that ESMF is fine with longitudes in either the -180 to 180 range or the 0 to 360 range.  I don't know much about Lilac or where that error is coming from. However, could you just convert the longitudes in memory after you read them in from the file or when they they are being passed into the component where the error is occurring? If that's not an option, ESMF doesn't currently have a tool to change the coordinates in a file. but you could try the [NCO ](https://nco.sourceforge.net/#Definition) set of NetCDF manipulation tools. There might be a way to do it using those. Let me know if you have further questions. Thanks!
